src/orderly/run.py

```diff
@@ -56,6 +62,36 @@ def _validate_src_directory(name, root):
     return path
 
 
+def _validate_parameters(given, defaults):
+    if given is None:
+        given = {}
+
+    if given and not defaults:
+        msg = "Parameters given, but none declared"
+        raise Exception(msg)
+
+    required = [k for k, v in defaults.items() if v is None]
+    missing = set(required).difference(given.keys())
```

Comment:
```suggestion
    required = {k for k, v in defaults.items() if v is None}
    missing = required.difference(given.keys())
```

Using a set comprehension, instead of a list comprehension followed by a `set()` call.